base/shell/explorer/traywnd.cpp

```diff
@@ -3262,16 +3284,23 @@ class CTrayWindow :
                 return TRUE;
         }
 
-        if (::IsWindowVisible(hwnd) && !::IsIconic(hwnd))
+        if (ShouldMinimize(hwnd))
         {
-            MINWNDPOS mwp;
-            mwp.hwnd = hwnd;
-            mwp.wndpl.length = sizeof(mwp.wndpl);
-            ::GetWindowPlacement(hwnd, &mwp.wndpl); // Save the position and status
-
-            info->pMinimizedAll->Add(mwp);
-
-            ::ShowWindowAsync(hwnd, SW_MINIMIZE);
+            MINWNDPOS mwp = { hwnd, { sizeof(mwp.wndpl) } };
+            if (::GetWindowPlacement(hwnd, &mwp.wndpl) && // Save the position and status
+                ::ShowWindowAsync(hwnd, SW_SHOWMINNOACTIVE)) // Minimize
+            {
+                info->pMinimizedAll->Add(mwp);
+            }
+        }
+        else if (CanDialogSysMinimize(hwnd))
+        {
+            MINWNDPOS mwp = { hwnd, { sizeof(mwp.wndpl) } };
+            if (::GetWindowPlacement(hwnd, &mwp.wndpl) && // Save the position and status
+                ::PostMessageW(hwnd, WM_SYSCOMMAND, SC_MINIMIZE, -1)) // Minimize
```

Comment:
Doesn't `::ShowWindowAsync(hwnd, SW_SHOWMINNOACTIVE)` work for dialog boxes as well?

Reply:
I'm not sure why it is this code... But it is not `ShouldMinimize`.

Reply:
Do you mean that doing:
```c
        if (ShouldMinimize(hwnd) || CanDialogSysMinimize(hwnd))
        {
            MINWNDPOS mwp = { hwnd, { sizeof(mwp.wndpl) } };
            if (::GetWindowPlacement(hwnd, &mwp.wndpl) && // Save the position and status
/*** Here, either just using that ShowWindowAsync, or the PostMessage version only ***/
                ::ShowWindowAsync(hwnd, SW_SHOWMINNOACTIVE)) // Minimize
            {
                info->pMinimizedAll->Add(mwp);
            }
        }
```
wouldn't work?

Reply:
Deleted `CanDialogSysMinimize`.